### ISSUE TITLE
Collapse sidebar on narrow screens

### ISSUE THREAD

**Issue by waldyrious** [openstreetmap/iD#9815]
### Description

When opening iD in a device with a narrow screen, the sidebar appears automatically even though it is empty, occupying precious real estate:

![Screen Shot 2023-08-05 at 02 08 28](https://github.com/openstreetmap/iD/assets/478237/a0433807-0e95-4cf3-ab43-4ad913c06951)

IMO it should automatically be loaded in the collapsed state when the container is narrow:

![Screen Shot 2023-08-05 at 02 10 13](https://github.com/openstreetmap/iD/assets/478237/24952019-6cb1-4035-99ec-c4e0c711fffd)

As an example of existing behavior that matches the expectation, the toolbar buttons already  adopt a more compact display in narrower screens by omitting the labels and showing just the icons, which IMO is sensible behavior:

| Wide screen | Narrow screen |
| ------ | ----- |
| ![image](https://github.com/openstreetmap/iD/assets/478237/86bc922b-7e16-4841-ad69-d35d0f72305f) | ![image](https://github.com/openstreetmap/iD/assets/478237/3ac9cab6-8a2b-40ad-9a48-c67e4f114ed4) |

I would gladly contribute a PR making the sidebar also auto-collapse in narrow screens, if there's agreement that this is desirable, and if I can get some guidance on how to do it — I suppose one needs to do something like, say `selection.classed('collapsed', containerWidth < 600);` (or whatever pixel width we consider narrow enough) somewhere when [initializing the sidebar](https://github.com/openstreetmap/iD/blob/5a14edfe368e14e7adf359d036ce5fe3fb1140be/modules/ui/sidebar.js#L38)?

**Comment by AviralSingh-code** [comment]
Hello !!
I think my PR will be able to solve the issue !!

**Comment by tyrasd** [comment]
It could be confusing for users when the sidebar is (sometimes) completely collapsed: In order to edit the properties one must now, after selecting the feature, also click on the top-left button to open the sidebar again.

I think it would be better to revisit the UI concept from #7496: essentially just removing the gray search box as long as no search results are to be displayed (and no OSM feature is selected) should make the map much more usable on narrow screens.

**Comment by AlbertoFabbri93** [comment]
I agree the behaviour should be kept consistent between mobile and desktop and I agree that hiding the sidebar when empty makes sense. Honestly I find quite confusing having a big empty sidebar as its purpose is not immediately clear.
